### ISSUE TITLE
Global styles navigation button: avoid visible text and accessible name mismatch

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#66526]
### Description

Similar to https://github.com/WordPress/gutenberg/issues/66525

In the Site editor 'site view', the navigation panel > Styles, shows a control to go to the Revisions:

![Image](https://github.com/user-attachments/assets/df323987-1ebd-4950-8330-e43147ef40bb)

The visible text of this button is, for example:

`Last modified 3 days ago`

the actual accessible nae provided by the button aria-label is:

`aria-label="Revisions"`

FOr the same reasons explained in https://github.com/WordPress/gutenberg/issues/66525, this kinf od mismatch must be avoided.

Also, interactive controls aren't the right place to provide information about states or values. They have to provide information about the button action.

### Step-by-step reproduction instructions

- Make sure you have some global styles revisions. If necessary, go to the Site Editor > Styles, make any change and save.
- Go to the main navigation panel in the Site editor 'site view'.
- Go to the Styles sub-panel.
- Inspect the source of the button at the bottom of the panel.
- Observe the visible text is about the last modified date, while the accessible name is 'Revisions'.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [ ] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by up1512001** [comment]
@afercia can you please review this PR? 
https://github.com/WordPress/gutenberg/pull/66535/files

**Comment by afercia** [comment]
Thinking a bit more into this, I think the best way to improve usability, accessibility, and consistency is by separating the two informations: action and state should be communicated separately. That requires a design change.

Functionally, this button is a navigation button, as in: users click on it and the editor switches to the revisions view. This action is best communicated with a _visible label_. Only adding an `aria-label="Revisions"' helps screen reader users but isn't helpful for all other users. I would say this is a good example of how _not_ to use `aria-label` as the visible text completely mismatches the actual button name and underlying action.

The state `Last modified + humanTimeDiff` is a valuable information but buttons aren't the right place to communicate states or values.

In this navigation panel there are other examples of 'navigation buttons' that bring users to specific views of the editor. However, in most of the cases, the open a sub-panel of the 'drill-down' menu. That's the reason why the use a chevron icon. Screenshot to illustrate:

![Image](https://github.com/user-attachments/assets/6035dafb-814c-44f4-a66c-5dcd1001609b)

I'm not sure a chevron icon would be appropriate in this case because this button does not open a sub-panel of the drill-down menu. To my understanding, the usage of the chevron icon was specifically meant to indicate the drill-down menu mechanism. I can't think of other buttons that directly switch the editor to a specific view other than the vairous 'Edit' buttons in the navigation menu, which use a completely different pattern.

It is also worth mentioning that the `humanTimeDiff` string mauy wrap in two lines when it's longer. Ideally this should be prevented. Screenshot to illustrate:

![Image](https://github.com/user-attachments/assets/71bcae43-352c-45a3-a55c-d7c1e751a61d)

Overall, I'd lean towards making things as simple as possible. Something along the lines of the screenshots below could work. With and without chevron icon:

![Image](https://github.com/user-attachments/assets/be4a44ab-7d33-4aa7-a359-563017d9b345)

I'll submit a new PR as an alternative to https://github.com/WordPress/gutenberg/pull/66535
